cogs/MessageCounter.py

```diff
@@ -0,0 +1,45 @@
+import sqlite3
+
+import discord
+from discord.ext import commands
+
+
+class MessageCounter(commands.Cog):
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+        self.db = sqlite3.connect("util/database.sqlite")
+        self.cursor = self.db.cursor()
+        # if table does not exist, create it
+        self.cursor.execute(
+            """CREATE TABLE IF NOT EXISTS messagecount (
+                uid INTEGER PRIMARY KEY, 
+                amount INTEGER
+                )"""
+        )
+
+    @commands.Cog.listener()
+    async def on_message(self, message) -> None:
+        if message.author.bot:
+            return
+
+        amount = self.cursor.execute(
+            "SELECT amount FROM messagecount WHERE uid = ?", (message.author.id,)
+        ).fetchone()
+        if amount is None:
+            self.add_user(message.author.id)
+        else:
+            self.update_user(message.author.id, amount[0] + 1)
+
+    def add_user(self, uid: int) -> None:
+        self.cursor.execute("INSERT INTO messagecount VALUES (?, ?)", (uid, 1))
+        self.db.commit()
+
+    def update_user(self, uid: int, amount: int) -> None:
+        self.cursor.execute(
+            "UPDATE messagecount SET amount = ? WHERE uid = ?", (amount, uid)
+        )
+        self.db.commit()
```

Comment:
btw this could be a single query, since sqlite supports [upsert](https://www.sqlite.org/lang_UPSERT.html)
```py
self.cursor.execute("""
    INSERT INTO
        messagecount (uid, amount)
    VALUES
        (?, 1)
    ON CONFLICT(uid) DO
    UPDATE SET
        amount = amount + 1
    WHERE
        uid = ?
    """,
    (uid, uid)
)
```